launchlib/config.go

```diff
@@ -20,51 +20,56 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+type LauncherConfig struct {
+	ConfigType    string `yaml:"configType"`
+	ConfigVersion int    `yaml:"configVersion"`
+}
+
+func (config *LauncherConfig) validate() {
+	if config.ConfigType != "java" {
+		panic(fmt.Sprintf("Can handle configType=java only, found %s", config.ConfigType))
+	}
+	if config.ConfigVersion != 1 {
+		panic(fmt.Sprintf("Can handle configVersion=1 only, found %d", config.ConfigVersion))
+	}
+}
+
 type StaticLauncherConfig struct {
-	ConfigType    string            `yaml:"configType"`
-	ConfigVersion int               `yaml:"configVersion"`
-	ServiceName   string            `yaml:"serviceName"`
-	MainClass     string            `yaml:"mainClass"`
-	JavaHome      string            `yaml:"javaHome"`
-	Env           map[string]string `yaml:"env"`
-	Classpath     []string
-	JvmOpts       []string `yaml:"jvmOpts"`
-	Args          []string
+	LauncherConfig `yaml:",inline"`
+	ServiceName    string            `yaml:"serviceName"`
+	MainClass      string            `yaml:"mainClass"`
+	JavaHome       string            `yaml:"javaHome"`
+	Env            map[string]string `yaml:"env"`
+	Classpath      []string
+	JvmOpts        []string `yaml:"jvmOpts"`
```

Comment:
This is the alignment written/enforced by `gofmt` (Go's standard formatting tool).

Alignment can be made to look more natural by either moving declaration of `JvmOpts` before `ClassPath` or by adding `yaml` definitions for `ClassPath` and `Args` for consistency (would recommend the latter)